nekoyume/battle/logger.py

```diff
@@ -1,35 +1,30 @@
+from dataclasses import asdict
+import json
+
+
 class Logger:
     def __init__(self):
         self.logs = []
         self.print = False
 
-    def log_attack(self, dict_):
-        dict_['type'] = 'attack'
-        self.log(dict_)
-
-    def log_heal(self, dict_):
-        dict_['type'] = 'heal'
-        self.log(dict_)
-
-    def log_dead(self, dict_):
-        dict_['type'] = 'dead'
-        self.log(dict_)
-
-    def log_exp(self, exp):
-        dict_ = {
-            'type': 'get_exp',
-            'exp': exp,
-        }
-        self.log(dict_)
-
-    def log_item(self, item):
-        dict_ = {
-            'type': 'get_item',
-            'item': item,
-        }
-        self.log(dict_)
-
     def log(self, log):
         self.logs.append(log)
         if self.print:
             print(log)
+
+    def json_dump(self):
+        status = []
+        for log in self.logs:
+            status.append(asdict(log))
+        return json.dumps({'status': status})
+
+    @property
+    def get_characters(self):
```

Comment:
As it's not a method but a `property`, simply `characters` would be a better name than `get_characters`.

Reply:
That means there is a cost to find characters.
find_characters or get_characters?

Reply:
For that purpose, it'd better to be a vanilla method rather than `property`, i.e., `get_characters` → `get_characters()`.

Reply:
OK